src/components/table/table.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+
+const normalizeKey = (label) =>
+  label.trim().replace(/\s+/g, '').toLowerCase();
+
+const Table = ({
+  title,
+  headers = '',
+  rows = '',
+}) => {
+  const parsedHeaders = useMemo(() => {
+    return headers
+      .split(',')
+      .map((h) => h.trim())
+      .filter((h) => h)
+      .map((label) => ({
+        label,
+        key: normalizeKey(label),
+      }));
+  }, [headers]);
+
+  const bodyRows = useMemo(() => {
+    return rows
+      .split('~')
+      .map((rowStr) =>
+        rowStr
+          .split(',')
+          .map((c) => c.trim())
+      )
+      .map((cells) => {
+        const obj = {};
+        parsedHeaders.forEach((h, idx) => {
+          obj[h.key] = cells[idx];
+        });
+        return obj;
+      });
+  }, [rows, parsedHeaders]);
+
+  const renderCell = (row, h) => {
+    let cell = row[h.key];
+    return cell ?? "-";
+  };
```

Comment:
Looks great! I'd just maybe change the `one letter` variables to something else to make it more readable 🙂